### PR TITLE
[12.0][ADD] hr_timesheet_sheet_role: glue between hr_timesheet_role and hr_timesheet_sheet

### PR DESCRIPTION
**Pull request by alexey-pelykh**

### CODE REVIEW

**Comment by alexey-pelykh** [comment]
Depends on #140, @bjeficent @aheficent please review - likely you're the most aware of the topic

**Comment by alexey-pelykh** [comment]
Depends on #155

File: hr_timesheet_role/models/account_analytic_line.py
Comment (bjeficent) [inline hr_timesheet_role/models/account_analytic_line.py]: This field does not exist. I think that this field is in model ```hr_timesheet_sheet.line```.

File: hr_timesheet_role/models/account_analytic_line.py
Comment (bjeficent) [inline hr_timesheet_role/models/account_analytic_line.py]: Same as above.

File: hr_timesheet_role/models/account_analytic_line.py
Comment (alexey-pelykh) [reply hr_timesheet_role/models/account_analytic_line.py]: This PR is based on #140 (and has commits of it), so please ignore the `hr_timesheet_role` part

File: hr_timesheet_role/models/account_analytic_line.py
Comment (alexey-pelykh) [reply hr_timesheet_role/models/account_analytic_line.py]: This PR is based on #140 (and has commits of it), so please ignore the `hr_timesheet_role` part

**Comment by AaronHForgeFlow** [comment]
Hi @alexey-pelykh Can you include the other requirements in the oca_dependencies file? It is really hard to test this way.

We did all the merges of all the dependencies in local but we got many conflicts. We manually resolved them but not sure if we did correctly. Don't understand this: https://github.com/OCA/hr-timesheet/pull/154#discussion_r240520381

**Comment by alexey-pelykh** [comment]
@aheficent this PR depends on `hr_timesheet_role` & `hr_timesheet_sheet`, as far as I know, nothing extra should be added to `oca_dependencies.txt` - both dependencies are within this same repository. Once #140 and #155 are merged, I'll rebase this one and it will be as usual. Or I did not get you correctly

**Comment by alexey-pelykh** [comment]
yet what is important - _modified_ version of hr_timesheet_sheet

File: hr_timesheet_role/models/account_analytic_line.py
Comment (alexey-pelykh) [reply hr_timesheet_role/models/account_analytic_line.py]: Yet you where correct that in hr_timesheet_sheet there are no `project_id` - so thanks a lot for a review!

File: hr_timesheet_role/models/account_analytic_line.py
Comment (alexey-pelykh) [reply hr_timesheet_role/models/account_analytic_line.py]: Yet you where correct that in hr_timesheet_sheet there are no `project_id` - so thanks a lot for a review!

**Comment by AaronHForgeFlow** [comment]
@alexey-pelykh Yes, I know it should not me added as a dependency forever. I meant just temporary for testing on runbot. But nevermind it is ok. 

What we did to reproduce on local:

> checkout 12.0-add-hr_timesheet_role
> merge 12.0-imp-hr_timesheet_sheet-extendability
> merge 12.0-add-hr_timesheet_sheet_role

Got many conflicts on last merge. After resolve manually could not install the module cause of https://github.com/OCA/hr-timesheet/pull/154#discussion_r240519760 

Did we do something wrong?

**Comment by alexey-pelykh** [comment]
@aheficent at the moment it's enough just to checkout `12.0-add-hr_timesheet_sheet_role` branch - it has all needed commits within it

**Comment by alexey-pelykh** [comment]
@aheficent did that work for you?

**Comment by AaronHForgeFlow** [comment]
Yes it did. It works. But this is modifying the base module a lot and I am wondering if it is possible to do everything in a third module.

**Comment by alexey-pelykh** [comment]
Well, it's a glue module - it _is_ that third module

**Comment by alexey-pelykh** [comment]
Btw, `hr_timesheet_sheet` modification is actually #155

**Comment by AaronHForgeFlow** [comment]
Opps yes. Sorry ,i did not realize you included the commits from the other PR here.

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: Please wait until the other PR is merged and then rebase this PR.

**Comment by alexey-pelykh** [comment]
No worries 👍 That's temporary until #155 is merged

**Comment by alexey-pelykh** [comment]
@aheficent @bjeficent #155 has been merged, this PR has been rebased, thus it's ok proceed with this one

**Review by bjeficent — APPROVED** [review]
LGTM :+1:

**Comment by alexey-pelykh** [comment]
@pedrobaeza some magic needed, if you have a spare minute 👍

**Comment by alexey-pelykh** [comment]
@dreispt if you don't mind I'd like to ask you to take a look on this PR as PSC

File: hr_timesheet_sheet_role/models/hr_timesheet_sheet.py
Comment (lmignon) [inline hr_timesheet_sheet_role/models/hr_timesheet_sheet.py]: @alexey-pelykh Why not `user_id = self.user_id`? IMO `onchange_add_project_id` should also depend on `user_id` otherwise the domain will not be recomputed if the employee id is changed on the timesheet. `https://github.com/OCA/hr-timesheet/blob/12.0/hr_timesheet_sheet/models/hr_timesheet_sheet.py#L78`

Comment (alexey-pelykh) [reply hr_timesheet_sheet_role/models/hr_timesheet_sheet.py]: Valid point, fixed!

**Comment by alexey-pelykh** [comment]
@aheficent @bjeficent @lmignon if possible, please take a look on https://github.com/OCA/timesheet/pull/187 as this PR is based on it

**Comment by alexey-pelykh** [comment]
This PR is good to be merged, as dependencies are already merged into branch.

**Pull request merged**
